minHeap/minHeap.go

```diff
@@ -0,0 +1,95 @@
+package heap
+
+import (
+	"errors"
+	"fmt"
+)
+
+type heap struct {
+	store    []int
+	capacity int
+	size     int
+}
+
+//creates a heap from an int array
+func buildMinHeap(values []int) (*heap, error) {
+	if len(values) == 0 {
+		return nil, errors.New("Array is empty")
+	}
+	minHeap := heap{capacity: (len(values) * 2), size: 0}
+	h := &minHeap
+	for _, value := range values {
+		h.insert(value)
+		fmt.Println(h.store)
+	}
+	return h, nil
+}
+
+// checks if the array is not full
+func (h *heap) checkCapacity() {
```

Comment:
is this the way to double the size?

Reply:
if the capacity is full, a new array is created with double the size of the original array.